---
title: "Frame Focusable Content"
description: "Frames with focusable content must not have tabindex=\"-1\""
---

import { Callout } from 'nextra/components'
import { CTA2 } from '@components/mdx/HelpUI'

# Frame Focusable Content

<Callout type="info">
Frames with focusable content must not have tabindex="-1"
</Callout>

Frames containing focusable content must not use tabindex="-1", because doing so prevents users from navigating into them using the keyboard. This blog explains what the frame-focusable-content rule checks, why incorrect tabindex values block access to embedded content, how to correct frame accessibility issues and how this supports WCAG 2.2 and wider compliance expectations. The article is fully original, SEO‑optimised and structured using the Welcoming Web content framework.

## What it is

The frame-focusable-content rule checks whether &lt;frame&gt; or &lt;iframe&gt; elements containing interactive or focusable content are assigned tabindex="-1". The value tabindex="-1" removes the element from the natural tab order. When applied to a frame that contains controls, form fields or focusable widgets, users cannot tab into the embedded content.

Common causes include: - using negative tabindex values to hide frame focus for styling reasons, - incorrect template defaults in CMS output, - embedded widgets that rely on scripted focus management, - copied iframe snippets from third‑party providers that include restrictive tabindex values.

## Why it matters

Frames often contain rich, interactive experiences such as videos, maps, forms or embedded applications. When users cannot tab into these frames: - keyboard users are blocked from accessing the content inside, - assistive technologies may skip important controls, - interaction becomes unpredictable and frustrating, - tasks such as form completion or media playback become impossible.

Removing inappropriate tabindex="-1" values supports predictable navigation and ensures embedded content remains operable.

## Who delivers it

Front end developers verify frame tabindex values and ensure frames remain reachable. Designers confirm that embedded content behaves consistently in keyboard navigation. Accessibility specialists and QA testers validate frame behaviour across assistive technologies. Welcoming Web assists by detecting frames with focusable content that are incorrectly removed from the tab order.

How to fix frames with focusable content using tabindex="-1"

1. Remove tabindex="-1" from frames containing interactive elements

Allow users to tab naturally into the embedded content.

### Incorrect example:

```html
<iframe src="map.html" tabindex="-1"></iframe>
```

### Corrected version:

```html
<iframe src="map.html"></iframe>
```

2. Ensure focus moves correctly into embedded applications

Test keyboard navigation to confirm users can enter and leave the frame.

3. Avoid scripted focus suppression

Do not override focus behaviour inside or around frames unless necessary.

4. Review third‑party embed snippets

Remove restrictive tabindex values added by external providers.

5. Validate dynamic frame insertion

Ensure that frames generated by JavaScript do not default to negative tabindex values.

## Best practice guidance

Embed content that behaves predictably for all users. Avoid hiding focus for visual reasons. Provide clear keyboard access to controls inside frames, especially for video players, maps, authentication widgets or embedded forms. Test embedded content with multiple assistive technologies.

## Compliance mapping

Ensuring frames remain keyboard accessible supports: - WCAG 2.2 requirements for Keyboard Accessibility and Focus Order, - WCAG 2.2 Name, Role, Value expectations for embedded controls, - ADA Title III expectations for operable content inside embedded applications, - EN 301 549 guidance for device‑independent input, - Equality Act 2010 duties for accessible, consistent navigation.

Welcoming Web supports alignment with recognised standards but does not issue or guarantee compliance certification.

## How Welcoming Web supports teams

Welcoming Web detects &lt;frame&gt; and &lt;iframe&gt; elements containing focusable content that are incorrectly assigned tabindex="-1". The platform provides guidance to restore keyboard access and improve the usability of embedded content.

## Key points for development teams

Frames with interactive content must be focusable. Do not use tabindex="-1" on embedded content. Check third‑party embed snippets. Test keyboard entry and exit behaviour. Maintain predictable navigation paths.

## Call to action

Run an audit Check your site for frames that block keyboard access. Supports WCAG 2.2 and ADA goals.

## FAQs

What does the frame-focusable-content rule check

It checks whether frames with focusable content incorrectly use tabindex="-1".

Why is tabindex="-1" a problem on frames

It removes the frame from the tab order and prevents users from accessing interactive content.

How do I fix this issue

Remove the negative tabindex value and test keyboard navigation.

Can third‑party embeds cause this

Yes. Some embed codes use restrictive tabindex attributes.

Does fixing frame tabindex issues improve accessibility

Yes. It ensures keyboard and assistive technology users can access embedded content.

Does this guarantee WCAG compliance

It supports keyboard operability but does not guarantee full compliance.

How does Welcoming Web help

Welcoming Web detects frames removed from the tab order and offers guidance for correction.

Disclaimer

Welcoming Web supports accessibility improvement and alignment with recognised standards but does not issue or guarantee compliance certification.

<CTA2
  title="Need More Help?"
  description="Schedule a personal support session or join our live training webinars."
  buttonText="Contact Support"
  buttonLink="https://welcomingweb.com/contact"
/>
