---
title: "Landmark Contentinfo Is Top Level"
description: "Contentinfo landmark should not be contained in another landmark"
---

import { Callout } from 'nextra/components'
import { CTA2 } from '@components/mdx/HelpUI'

# Landmark Contentinfo Is Top Level

<Callout type="info">
Contentinfo landmark should not be contained in another landmark
</Callout>

The contentinfo landmark must appear at the top level of the document so users can understand where global information and footer content belong. This blog explains what the landmark-contentinfo-is-top-level rule checks, why correct placement is essential for navigability, how to ensure valid landmark hierarchy and how this supports WCAG 2.2 and wider compliance goals. The article is fully original, accurate and structured using the Welcoming Web content framework.

## What it is

The landmark-contentinfo-is-top-level rule checks whether the contentinfo landmark—most often represented by the &lt;footer&gt; element—is placed at the top level of the page rather than nested inside other landmarks. Contentinfo is meant to provide site-wide information such as copyright details, legal notices, contact information and persistent global content.

If contentinfo is nested inside other landmarks like &lt;main&gt;, &lt;header&gt;, &lt;nav&gt; or &lt;aside&gt;, assistive technologies may misrepresent the page structure and cause confusion.

## Why it matters

Correct use of the contentinfo landmark helps screen reader users understand the end‑of‑page structure and locate global information. When the landmark is nested improperly: - the page outline becomes misleading, - users may not recognise footer content as consistent across pages, - navigation through landmarks becomes less predictable, - important global information may be interpreted as part of the main content, - assistive technologies may skip or miscategorise the footer.

Top‑level contentinfo placement supports predictable navigation and clearer understanding of page structure.

## Who delivers it

Front end developers control landmark placement through semantic HTML. Designers establish layout patterns that keep the footer at the correct structural level. Content authors ensure footer content is appropriate for a global region. Accessibility specialists and QA testers verify correct hierarchy across templates. Welcoming Web assists by detecting contentinfo landmarks placed incorrectly within other regions.

How to ensure the contentinfo landmark is top‑level

1. Place the footer directly under &lt;body&gt;

Avoid nesting the contentinfo region inside &lt;main&gt;, &lt;nav&gt;, &lt;header&gt; or &lt;aside&gt;.

### Correct example:

```html
<body>  <header>…</header>  <main>…</main>  <footer role="contentinfo">Global site information</footer></body>
```

2. Use semantic HTML

Use &lt;footer&gt; for site‑wide footers and allow the browser to apply the contentinfo role automatically.

3. Do not create multiple contentinfo landmarks unnecessarily

Only one site‑wide contentinfo region should exist unless there are well‑defined separate contexts.

4. Keep layout consistent across templates

Ensure reusable components and CMS theme structures do not incorrectly nest footer regions.

5. Validate with assistive technology tools

Use browser landmark views or screen readers to confirm the footer appears at the expected hierarchy level.

## Best practice guidance

Keep global footer content short and meaningful. Include links to policies, contact details or legal information. Avoid placing interactive regions or complex layouts inside the footer unless they remain semantically clear. Document landmark usage within your design system to maintain structural consistency.

## Compliance mapping

Correct placement of the contentinfo landmark supports: - WCAG 2.2 Info and Relationships success criteria, - WCAG 2.2 Navigable expectations for region clarity, - ADA Title III requirements for predictable structure, - EN 301 549 guidance on programmatically determinable regions, - Equality Act 2010 duties for accessible information presentation.

Welcoming Web supports alignment with recognised standards but does not issue or guarantee compliance certification.

## How Welcoming Web supports teams

Welcoming Web identifies footer regions that are incorrectly nested inside other landmarks. The platform highlights structural issues and provides guidance for achieving valid, predictable landmark hierarchy.

## Key points for development teams

Contentinfo must be top‑level. Use one global footer per page. Do not nest contentinfo inside other regions. Use semantic &lt;footer&gt; markup. Validate templates and CMS structures.

## Call to action

Run an audit Check your site for incorrectly nested contentinfo landmarks. Supports WCAG 2.2 and ADA goals.

## FAQs

What does the landmark-contentinfo-is-top-level rule check

It checks whether the contentinfo landmark or &lt;footer&gt; element is placed at the top level instead of inside other landmarks.

Why must the footer be top‑level

It must be top‑level so users can clearly identify global site information and navigate the page structure predictably.

Can I have multiple footers

Only one global footer should use the contentinfo role unless the page has distinct, separate regions that require additional context.

What happens if contentinfo is nested

Nested contentinfo landmarks confuse screen reader users and disrupt the page outline.

Does &lt;footer&gt; always map to contentinfo

Yes, &lt;footer&gt; used at the top level automatically represents contentinfo.

Do sub‑section footers use contentinfo

No. Footers inside articles or sections should not use the contentinfo role.

Does correct footer placement guarantee WCAG compliance

It supports structure and navigation requirements but does not guarantee full compliance.

How does Welcoming Web help with contentinfo issues

Welcoming Web detects nested contentinfo landmarks and provides guidance to correct structural placement.

Disclaimer

Welcoming Web supports accessibility improvement and alignment with recognised standards but does not issue or guarantee compliance certification.

<CTA2
  title="Need More Help?"
  description="Schedule a personal support session or join our live training webinars."
  buttonText="Contact Support"
  buttonLink="https://welcomingweb.com/contact"
/>
