---
title: "Link Name"
description: "Links must have discernible text"
---

import { Callout } from 'nextra/components'
import { CTA2 } from '@components/mdx/HelpUI'

# Link Name

<Callout type="info">
Links must have discernible text
</Callout>

Links must include discernible text so users can understand their purpose before activating them. This blog explains what the link-name rule checks, why clear link names are essential for accessibility, how to provide meaningful link text and how this supports WCAG 2.2 and wider compliance expectations. The article is fully original, accurate and structured using the Welcoming Web content framework.

## What it is

The link-name rule checks whether each link exposes a discernible and meaningful name. Links need accessible text that communicates their destination or action. This name may come from visible link text or, when appropriate, accessible naming techniques such as aria-label or aria-labelledby.

Links without discernible names are announced as “link” with no context, making navigation confusing and unpredictable.

## Why it matters

When links lack discernible text: - screen reader users cannot understand what the link does, - users may activate the wrong link or fail to complete tasks, - navigation becomes inefficient, especially when tabbing through multiple links, - users with cognitive impairments may struggle to interpret context, - assistive technologies cannot present meaningful lists of links.

Clear link names support predictable navigation and reduce user effort.

## Who delivers it

Content authors write descriptive link text. Designers ensure links remain visible and clear. Front end developers implement correct accessible names using semantic HTML and ARIA when needed. Accessibility specialists and QA testers validate link names across templates and components. Welcoming Web assists by detecting links without discernible text.

How to ensure links have discernible text

1. Provide clear visible text for every link

Visible text should indicate the link’s purpose.

### Correct example:

```html
<a href="/pricing">View pricing</a>
```

2. Do not use empty or generic text

Avoid labels such as “click here”, “read more” or empty anchors.

3. Use ARIA naming only when necessary

For icon-only links, ensure ARIA provides an accessible name.

```html
<a href="/cart" aria-label="View shopping cart">  <svg>…</svg></a>
```

4. Avoid redundant or hidden text patterns

Ensure CSS does not visually hide text incorrectly or clip important link content.

5. Review dynamic link generation

Templates or scripts must include accessible names for all generated links.

## Best practice guidance

Write link text that describes the destination or action. Use action-focused phrases such as “Download guide” or “Manage account”. Maintain consistent naming patterns across navigation menus, cards and content sections. Avoid repeating link text that does not reflect the actual destination.

## Compliance mapping

Providing discernible link text supports: - WCAG 2.2 Name, Role, Value requirements, - WCAG 2.2 Link Purpose (In Context) success criteria, - ADA Title III expectations for understandable navigation, - EN 301 549 guidance on programmatically determinable names, - Equality Act 2010 duties for accessible communication.

Welcoming Web supports alignment with recognised standards but does not issue or guarantee compliance certification.

## How Welcoming Web supports teams

Welcoming Web detects links that do not expose clear, descriptive accessible names. The platform highlights empty anchors, icon-only links without labels and patterns that result in missing or unclear link text.

## Key points for development teams

Every link needs a discernible name. Visible text should describe purpose. Use ARIA for icon-only links. Avoid generic labels. Validate dynamic link patterns.

## Call to action

Run an audit Check your site for links without meaningful text. Supports WCAG 2.2 and ADA goals.

## FAQs

What does the link-name rule check

It checks whether links expose a meaningful accessible name.

Why must links have descriptive text

They need descriptive text so users understand the purpose before activating the link.

Can icons act as links

Yes, but icon-only links must provide an accessible name using ARIA.

Why avoid “click here”

Such labels provide no context when read out of surrounding text.

Does invisible text count as a link name

Only if implemented correctly using accessible techniques. Hidden visual text must remain programmatically available.

Do dynamic links need names

Yes. All links, including generated ones, must include meaningful accessible names.

Does adding link text guarantee WCAG compliance

It supports naming and purpose requirements but does not guarantee full compliance.

How does Welcoming Web help with link naming

Welcoming Web identifies links missing meaningful names and provides guidance for correction.

Disclaimer

Welcoming Web supports accessibility improvement and alignment with recognised standards but does not issue or guarantee compliance certification.

<CTA2
  title="Need More Help?"
  description="Schedule a personal support session or join our live training webinars."
  buttonText="Contact Support"
  buttonLink="https://welcomingweb.com/contact"
/>
