---
title: "Presentation Role Conflict"
description: "Elements marked as presentational should be consistently ignored"
---

import { Callout } from 'nextra/components'
import { CTA2 } from '@components/mdx/HelpUI'

# Presentation Role Conflict

<Callout type="info">
Elements marked as presentational should be consistently ignored
</Callout>

Elements marked as presentational must be ignored by assistive technologies. This blog explains what the presentation-role-conflict rule checks, why role conflicts create accessibility barriers, how to structure presentational elements correctly and how this supports WCAG 2.2 and wider compliance expectations. The article is fully original, accurate and structured using the Welcoming Web content framework.

## What it is

The presentation-role-conflict rule checks whether an element assigned role="presentation" or role="none" also contains global ARIA attributes or a tabindex value. These attributes give the element semantic meaning or focusability, which conflicts with the instruction that the element is purely decorative.

When a presentational element receives semantic roles or focus behaviour, assistive technologies may interpret it incorrectly.

## Why it matters

Conflicting roles create multiple accessibility issues: - screen readers may not know whether to ignore or announce the element, - users may encounter focusable decorative items, disrupting navigation, - inconsistent semantics make interactive controls difficult to interpret, - redundant or meaningless information increases cognitive load, - assistive technologies may announce confusing or contradictory cues.

Marking an element as presentational means it must behave like a decorative element with no semantic value.

## Who delivers it

Front end developers ensure that decorative elements do not receive ARIA attributes or tabindex values. Designers define which elements are decorative and should be ignored by assistive technologies. Content authors maintain clarity by avoiding unnecessary ARIA usage in rich text. Accessibility specialists and QA testers validate that presentational roles do not conflict with interactive or semantic attributes. Welcoming Web assists by detecting these conflicts.

How to ensure presentational elements are consistently ignored

1. Use role="presentation" or role="none" correctly

Apply these roles only to decorative elements that convey no information.

2. Remove global ARIA attributes from presentational elements

Avoid attributes such as: - aria-label, - aria-labelledby, - aria-describedby, - interactive ARIA states.

3. Ensure presentational elements are not focusable

Remove tabindex values that allow users to tab to decorative items.

```html
<!-- Incorrect --><span role="presentation" tabindex="0"></span><!-- Correct --><span role="presentation"></span>
```

4. Check SVG and icon usage

Icons used purely for decoration must not expose ARIA attributes unless intentionally labelled.

5. Validate component libraries

Reusable components must distinguish clearly between decorative and semantic elements.

## Best practice guidance

Use presentational roles sparingly and only when the element exists exclusively for layout or decoration. When an element needs to convey meaning, remove presentational roles and apply appropriate semantics or ARIA attributes. Document usage patterns in design systems so decorative elements remain consistent across components.

## Compliance mapping

Avoiding role conflicts supports: - WCAG 2.2 Name, Role, Value success criterion, - WCAG 2.2 Meaningful Sequence requirements, - ADA Title III expectations for predictable semantics, - EN 301 549 guidance on consistent role usage, - Equality Act 2010 duties for clear communication.

Welcoming Web supports alignment with recognised standards but does not issue or guarantee compliance certification.

## How Welcoming Web supports teams

Welcoming Web detects elements with role="presentation" that include ARIA attributes or focus behaviour. The platform highlights structural conflicts and provides guidance for restoring consistent semantics.

## Key points for development teams

Do not add ARIA attributes to presentational elements. Avoid tabindex on decorative items. Ensure decorative icons do not expose semantic roles. Use presentational roles only when elements provide no meaning. Validate components for role conflicts.

## Call to action

Run an audit Check your site for presentational elements with conflicting roles. Supports WCAG 2.2 and ADA goals.

## FAQs

What does the presentation-role-conflict rule check

It checks whether elements marked as presentational also contain semantic or focus attributes.

Why must presentational elements be ignored by assistive technologies

They must be ignored so users do not encounter decorative or meaningless elements.

What causes a role conflict

Adding ARIA attributes or tabindex to a presentational element creates a conflict.

Can icons be presentational

Yes, if they serve no functional or informational purpose.

How do I fix role conflicts

Remove presentational roles from meaningful elements or remove ARIA attributes from decorative ones.

Does role=“none” behave like role=“presentation”

Yes. Both roles instruct assistive technologies to ignore the element.

Does fixing role conflicts guarantee WCAG compliance

It supports semantic requirements but does not guarantee full compliance.

How does Welcoming Web help with presentational conflicts

Welcoming Web flags elements where presentational roles conflict with ARIA or tabindex attributes.

Disclaimer

Welcoming Web supports accessibility improvement and alignment with recognised standards but does not issue or guarantee compliance certification.

<CTA2
  title="Need More Help?"
  description="Schedule a personal support session or join our live training webinars."
  buttonText="Contact Support"
  buttonLink="https://welcomingweb.com/contact"
/>
